themes-default/slim/views/partials/showheader.mako

```diff
@@ -10,64 +10,47 @@
-                <% lastSeason = season_results.pop(-1) %>
-            % endif
+        <template v-if="$route.name !== 'snatchSelection' && show.seasons && show.seasons.length >= 1">
             <div id="show-specials-and-seasons" class="pull-right">
-                <span class="h2footer display-specials">
-                % if season_special:
-                    Display Specials: <a class="inner" style="cursor: pointer;">${'Hide' if app.DISPLAY_SHOW_SPECIALS else 'Show'}</a>
-                % endif
+                <span class="h2footer display-specials" v-if="show.seasons.find(({ season }) => season === 0)">
+                    Display Specials: <a @click="toggleSpecials()" class="inner" style="cursor: pointer;">{{ config.layout.show.specials ? 'Hide' : 'Show' }}</a>
                 </span>
 
                 <div class="h2footer display-seasons clear">
                     <span>
-                    % if (len(season_results) > 14):
-                        <select id="seasonJump" class="form-control input-sm" style="position: relative">
-                            <option value="jump">Jump to Season</option>
-                        % for seasonNum in season_results:
-                            <option value="#season-${seasonNum["season"]}" data-season="${seasonNum["season"]}">${'Season ' + str(seasonNum["season"]) if int(seasonNum["season"]) > 0 else 'Specials'}</option>
-                        % endfor
-                        </select>
-                    % else:
-                        Season:
-                        % for seasonNum in season_results:
-                            % if int(seasonNum["season"]) == 0:
-                                <app-link href="#season-${seasonNum["season"]}">Specials</app-link>
-                            % else:
-                                <app-link href="#season-${seasonNum["season"]}">${str(seasonNum["season"])}</app-link>
-                            % endif
-                            % if seasonNum != season_results[-1]:
-                                <span class="separator">|</span>
-                            % endif
-                        % endfor
-                    % endif
+                        <template v-if="show.seasons && show.seasons.length >= 13">
+                            <select @change="jumpToSeason($event)" id="seasonJump" class="form-control input-sm" style="position: relative">
+                                <option value="jump">Jump to Season</option>
+                                <option v-for="season in show.seasons" :value="'#season-' + season.season" :data-season="season.season">
+                                    Season {{ season.season === 0 ? 'Specials' : season.season }}
+                                </option>
+                            </select>
+                        </template>
+                        <template v-else-if="show.seasons && show.seasons.length >= 1">
```

Comment:
You already checked for `show.seasons` above, so you know it's not `undefined` / `null`.
If you use my previous suggestion to use `<span>`, then you can also drop the `<template>` wrapper, and just put the `if` on the `<span>` element.

Even if you have to add another empty `<span v-else></span>` element, I still feel like it's much more readable.